### PR TITLE
Start parsing sysfs as well

### PR DESCRIPTION
**Pull request by dtrugman**
Based on issue #32, adding an initial version of a sysfs parser.
This was a good opportunity to spot some inconsistencies and do some housekeeping.
No API changes were required, so the new version is entirely backward compatible.

### CODE REVIEW

**Review by Alston-Tang — APPROVED** [review]
This is awesome! It covers the most important metrics in my mind. Thank you so much for implementing this. There might be something else like I'm using /<dev>/queue/rotational to check if this is a HDD or SSD. It should be straightforward to add later.

Not related to this but find_package doesn't seem to populate documented variables when I tested it. 
```
cmake_minimum_required(VERSION 3.16)

project(test)

find_package (pfs PATHS /home/alston64/works/pfs/build NO_DEFAULT_PATH REQUIRED)

include_directories ("/home/alston64/works/pfs/include")
add_executable(test test.cpp)
target_link_libraries(test "/home/alston64/works/pfs/build/lib/libpfs.a")
```
This was the cmake file I used. Basically I would expect to have `pfs_INCLUDE_DIRS` variable set if packaged is found in the given path but it was not the case. So I have to include header and library path manually.

File: src/parsers/block_stat.cpp [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+        utils::stot(tokens[DISCARD_IOS], stat.discard_ios, utils::base::decimal);
+        utils::stot(tokens[DISCARD_MERGES], stat.discard_merges, utils::base::decimal);
+        utils::stot(tokens[DISCARD_SECTORS], stat.discard_sectors, utils::base::decimal);
+        utils::stot(tokens[DISCARD_TICKS], stat.discard_ticks, utils::base::decimal);
+        utils::stot(tokens[FLUSH_IOS], stat.flush_ios, utils::base::decimal);
+        utils::stot(tokens[FLUSH_TICKS], stat.flush_ticks, utils::base::decimal);
+    }
+    catch (const std::invalid_argument& ex)
+    {
+        throw parser_error("Corrupted block stat - Invalid argument", line);
```
[the remark this answers is not in the document]
Comment (dtrugman) [reply src/parsers/block_stat.cpp:92]: You are right, this is something that could be improved all over.
I will create a ticket for that and try to address that asap.

**Comment by dtrugman** [comment]
> This is awesome! It covers the most important metrics in my mind. Thank you so much for implementing this. There might be something else like I'm using //queue/rotational to check if this is a HDD or SSD. It should be straightforward to add later.
> 
> Not related to this but find_package doesn't seem to populate documented variables when I tested it.
> 
> ```
> cmake_minimum_required(VERSION 3.16)
> 
> project(test)
> 
> find_package (pfs PATHS /home/alston64/works/pfs/build NO_DEFAULT_PATH REQUIRED)
> 
> include_directories ("/home/alston64/works/pfs/include")
> add_executable(test test.cpp)
> target_link_libraries(test "/home/alston64/works/pfs/build/lib/libpfs.a")
> ```
> 
> This was the cmake file I used. Basically I would expect to have `pfs_INCLUDE_DIRS` variable set if packaged is found in the given path but it was not the case. So I have to include header and library path manually.

Thanks for letting me know. I will open tickets for both (parsing rotational and fixing the `find_package` support)!

**Comment by dtrugman** [comment]
I will merge this and bump the version in the next few days.
I might wait with the bump to add support for a few more sysfs files and/or maybe trying to fix the other issues mentioned. Anyway, should be ready within a week or so.

**Pull request merged**